backend/de/api/v1.py

```diff
@@ -0,0 +1,333 @@
+import os
+from typing import Any, Dict, Iterable, List, Tuple
+
+import connexion
+import numpy as np
+import pandas as pd
+from ddtrace import tracer
+from flask import jsonify
+from scipy import stats
+from server_timing import Timing as ServerTiming
+
+from backend.common.utils.rollup import descendants
+from backend.wmg.api.wmg_api_config import (
+    WMG_API_FORCE_LOAD_SNAPSHOT_ID,
+    WMG_API_READ_FS_CACHED_SNAPSHOT,
+    WMG_API_SNAPSHOT_FS_CACHE_ROOT_PATH,
+    WMG_API_SNAPSHOT_SCHEMA_VERSION,
+)
+from backend.wmg.data.ontology_labels import gene_term_label, ontology_term_label
+from backend.wmg.data.query import DeQueryCriteria, WmgFiltersQueryCriteria, WmgQuery
+from backend.wmg.data.schemas.expression_summary_cube_schemas_diffexp import (
+    base_expression_summary_indexed_dims,
+    expression_summary_secondary_dims,
+)
+from backend.wmg.data.snapshot import WmgSnapshot, load_snapshot
+from backend.wmg.data.utils import (
+    find_all_dim_option_values,
+    find_dim_option_values,
+)
+
+DEPLOYMENT_STAGE = os.environ.get("DEPLOYMENT_STAGE", "")
+SNAPSHOT_FS_ROOT_PATH = (
+    WMG_API_SNAPSHOT_FS_CACHE_ROOT_PATH if (WMG_API_READ_FS_CACHED_SNAPSHOT and DEPLOYMENT_STAGE != "test") else None
+)
+
+
+@tracer.wrap(name="filters", service="wmg-api", resource="filters", span_type="wmg-api")
+def filters():
+    request = connexion.request.json
+    sanitize_api_query_dict(request["filter"])
```

Comment:
Discussion: what are the pros of mutating the original request object (~immutability principle), vs assigning a sanitized return `sanitized_filter = sanitize_api_query_dict(request["filter"])` and working off of that?

Reply:
Great question, don't think there are any pros beyond not needing a new variable - this is copied from the WMG API. Future work could refactor this logic into a shared utilities module.